Normalize check constraints even more radically

apollo13 opened the pull request:
This is the only way I could get this test pass on informix, basically I strip every whitespace.
The sql text as recorded in informix is:
```
((a > 1 ) AND (a < 5 ) )
((a = 1 ) OR ((a > 2 ) AND (a <5 ) ) )
```
Yes, this is absolutely bonkers, but that is what I get :(

zzzeek commented on lib/sqlalchemy/testing/suite/test_reflection.py:
```diff
@@ -789,19 +789,21 @@ def _test_get_check_constraints(self, schema=None):
+        # trying to minimize effect of quoting, parenthesis, etc.
+        # may need to add more to this as new dialects get CHECK
+        # constraint reflection support
+        normalize = lambda text: re.sub(r"[`'\(\)]", '', text.lower()).replace(' ', '')
```
this should be a def

zzzeek commented on lib/sqlalchemy/testing/suite/test_reflection.py:
```diff
@@ -789,19 +789,21 @@ def _test_get_check_constraints(self, schema=None):
             [
-                {'name': 'cc1', 'sqltext': 'a > 1 and a < 5'},
-                {'name': 'cc2', 'sqltext': 'a = 1 or a > 2 and a < 5'}
+                {'name': 'cc1', 'sqltext': normalize('a > 1 and a < 5')},
```
this should not be needed, normalize should remove all parens and convert groups of spaces into one space, right and left trim.  here's how to get that:

```
>>> sqltext = '((a = 1 ) OR ((a > 2 ) AND (a <5 ) ) )'
>>> " ".join(re.findall(r"and|\d|=|a|or|<|>", sqltext, re.I))
'a = 1 OR a > 2 AND a < 5'

```

apollo13 commented:
Ok, addressed the issues as suggested by you. I initially went for my solution since it keeps "more" of the original structure (ie if the check constraint has something completely different in it, but that shouldn't be the case either way).

zzzeek commented:
this works for you, right?   i will pull it into gerrit and if it works on my end too it gets merged.

zzzeek commented:
Dear contributor -

This pull request is being moved to Gerrit, at https://gerrit.sqlalchemy.org/#/c/zzzeek/sqlalchemy/+/637, where it may be tested and reviewed more closely.  As such, the pull request itself is being marked "closed" or "declined", however your contribution is merely being moved to our central review system. Please register at https://gerrit.sqlalchemy.org#/register/ to send and receive comments regarding this item.

apollo13 commented:
Last time I checked it worked, yes ;)
